src/test/unit/cli_unittest.cc

```diff
@@ -205,6 +222,283 @@ TEST(CLIUnittest, TestCliSetStringWriteOnce)
     EXPECT_EQ(0,   data[6]);
 }
 
+static uint8_t data[1000];
+static vector<string> outLines;
+
+class CliWriteTest : public ::testing::Test
+{
+
+protected:
+    static void SetUpTestCase() {}
+
+    virtual void SetUp() {
+        bufWriterInit(&cliWriterDesc, data, ARRAYLEN(data), &dummyBufWriter, NULL);
+        cliWriter = cliErrorWriter = &cliWriterDesc;
+    }
+
+    virtual void TearDown() {
+        cliWriter = NULL;
+        outLines.clear();
+    }
+
+    static void dummyBufWriter(void *, void *_data, int size) {
+        uint8_t *data = (uint8_t *)_data;
+        ostringstream stream;
+        for (int i = 0; i < size; i++) {
+            stream << (char)data[i];
+        }
+        outLines.push_back(stream.str());
+    }
+
+    void clear() {
+        outLines.clear();
+    }
+};
+
+class AuxCliWriteTest : public CliWriteTest {
+protected:
+    virtual void SetUp() {
+        CliWriteTest::SetUp();
+
+        for (uint8_t i = 0; i <= MAX_AUX_CHANNEL_COUNT; i++) {
+            memset(modeActivationConditionsMutable(i), 0, sizeof(modeActivationCondition_t));
+        }
+    }
+};
+
+// Help tests
+TEST_F(CliWriteTest, HelpAll)
+{
+    const char cmd[] = "help";
+    char args[] = "";
+    cliHelp(cmd, args);
+    EXPECT_LT(100, outLines.size());
+}
+
+TEST_F(CliWriteTest, HelpFindByName)
+{
+    const char cmd[] = "help";
+    char args[] = "aux";
+    cliHelp(cmd, args);
+    vector<string> expected = {
+        "aux", " - configure modes", "\r\n\t<index> <mode> <aux> <start> <end> <logic>", "\r\n",
+    };
+    EXPECT_EQ(expected, outLines);
+}
+
+TEST_F(CliWriteTest, HelpSearchByDescription)
+{
+    const char cmd[] = "help";
+    char args[] = "reb";
+    cliHelp(cmd, args);
+    vector<string> expected = {
+        "bl", " - reboot into bootloader", "\r\n\t[rom]", "\r\n",
+        "defaults", " - reset to defaults and reboot", "\r\n\t{nosave}", "\r\n",
+        "exit", " - exit command line interface and reboot (default)", "\r\n\t[noreboot]", "\r\n",
+        "save", " - save and reboot (default)", "\r\n\t[noreboot]", "\r\n",
+    };
+    EXPECT_EQ(expected, outLines);
+}
+// End of help tests
+
+// Aux tests
+TEST_F(AuxCliWriteTest, PrintAux_Default)
+{
+    const char heading[] = "aux";
+    printAux(DUMP_MASTER, modeActivationConditions(0), NULL, heading);
+    vector<string> expected = {
+        "\r\n# ", "aux", "\r\n",
+        "aux 0 0 0 900 900 0 0", "\r\n",
```

Comment:
This is a bit fragile - it is implementation detail that  "\r\n" is separated from rest of line (and extra flush is in code is degrading performance).

It should be quite easy to reparse lines `dummyBufWriter` (last vector element can contain line data visible so far). And when at it, maybe replace "\r\n" with standard C '\n'

Reply:
On the other side we could see another problem here which possibly should be fixed someday. I also was surprised by the output.

But I agree, I'll merge the output. It would be better to write another test exactly for printing logic.

Reply:
It's actually not a big problem for now - compared with write(char) cost, flushing write buffer is cheap. But with improved serial performance, it may be more important. 
And breaking unrelated unittests may prevent code changes